tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -462,6 +459,31 @@ public synchronized void forEach(ElementVisitor reader) throws IOException {
     }
   }
 
+  /**
+   * First reads header to update element counts and file length. Then Scans each element in the
+   * queue for its length. To be used for Integrity Scans to detect queue corruption.
+   *
+   * @return int[] of all segment lengths.
+   * @throws IOException if there is a problem scanning the file (assume corruption).
+   */
+  public synchronized int[] scanAll() throws IOException {
+    // Read header to update all our counts
+    readHeader();
```

Comment:
readHeader() has side effects. Probably a version of it with no side effects.

Reply:
Good idea.